Neos.Media/Classes/Domain/Repository/AssetRepository.php

```diff
@@ -138,7 +146,14 @@ public function countAll()
         $rsm = new ResultSetMapping();
         $rsm->addScalarResult('c', 'c');
 
-        $queryString = "SELECT count(persistence_object_identifier) c FROM neos_media_domain_model_asset WHERE dtype != 'neos_media_imagevariant'";
+        if ($this->entityClassName === Asset::class) {
+            $queryString = 'SELECT count(a.persistence_object_identifier) c FROM neos_media_domain_model_asset a WHERE ' . $this->getAssetVariantFilterClauseForDql('a');
+        } else {
+            $queryString = sprintf(
+                "SELECT count(persistence_object_identifier) c FROM neos_media_domain_model_asset WHERE dtype = '%s'",
+                strtolower(str_replace('Domain_Model_', '', str_replace('\\', '_', $this->entityClassName)))
```

Comment:
Can we move this "class name to dtype" into a named (static?) method?

Reply:
Also, maybe it would be more straight-forward to move the if condition into the `getAssetVariantFilterClause` method, since the condition now makes it harder to scan what the difference in the two query cases is. But we always do a `select count(persistence_object_identifier)` of course

Reply:
I'd rather leave that as is, TBH…

Reply:
Not gonna block it for that :) Looks good otherwise